lib/screens/alerts/alert.ex

```diff
@@ -352,13 +364,12 @@ defmodule Screens.Alerts.Alert do
 
   # HAPPENING NOW
   # defined as: some active period contains the current time
-  defp happening_now(%{active_period: aps}) do
-    now = DateTime.utc_now()
+  defp happening_now(%{active_period: aps}, now \\ DateTime.utc_now()) do
     if Enum.any?(aps, &in_active_period(&1, now)), do: 1, else: 0
   end
 
-  def happening_now?(alert) do
-    case happening_now(alert) do
+  def happening_now?(alert, now \\ DateTime.utc_now()) do
```

Comment:
Would it make sense to consolidate the `happening_now` logic into one function? I was thinking something like:
```elixir
def happening_now?(%{active_period: aps}, now \\ DateTime.utc_now()) do
  if Enum.any?(aps, &in_active_period(&1, now)), do: true, else: false
end
```

Reply:
Unfortunately I can't fully remove the other function—it was defined first and we need a numeric value in some places, a boolean value in others. I tried to clear up what's going on there, though.